src/uucore/src/lib/parser/parse_size.rs

```diff
@@ -65,13 +65,18 @@ impl<'parser> Parser<'parser> {
     /// # Examples
     ///
     /// ```rust
-    /// use uucore::parse_size::parse_size;
-    /// assert_eq!(Ok(123), parse_size("123"));
-    /// assert_eq!(Ok(9 * 1000), parse_size("9kB")); // kB is 1000
-    /// assert_eq!(Ok(2 * 1024), parse_size("2K")); // K is 1024
-    /// assert_eq!(Ok(44251 * 1024), parse_size("0xACDBK"));
+    /// use uucore::parse_size::Parser;
+    /// let parser = Parser {
+    ///     default_unit: Some("M"),
+    ///     ..Default::default()
+    /// };
+    /// assert_eq!(Ok(123 * 1024 * 1024), parser.parse("123M")); // M is 1024^2
+    /// assert_eq!(Ok(123 * 1024 * 1024), parser.parse("123")); // default unit set to "M" on parser instance
+    /// assert_eq!(Ok(9 * 1000), parser.parse("9kB")); // kB is 1000
+    /// assert_eq!(Ok(2 * 1024), parser.parse("2K")); // K is 1024
+    /// assert_eq!(Ok(44251 * 1024), parser.parse("0xACDBK")); // 0xACDB is 44251 in decimal
     /// ```
-    pub fn parse(&self, size: &str) -> Result<u64, ParseSizeError> {
+    pub fn parse(&self, size: &str) -> Result<u128, ParseSizeError> {
```

Comment:
Maybe we can make this generic over the number type? For example using `Unsigned` from `num_traits`? Then we don't need separate methods for `u64` and `u128`.

Reply:
yep, considered that, but discarded due to the 
1) we only need it for 2 variants - u64 and u128
2) there is no way to restrict it to only u64 and u128 (at compile time) even with that crate/trait
However, if you think it is worth the trouble - I can try to find a workable solution with generics .
Another thought / option - how about a macro instead of generics that unfolds into two implementations (one for u64 and one for u128)?

Reply:
Good points, I'll think about it for a bit :)

Reply:
@tertsdiepraam any updates/suggestions on refactoring this? Or can it be merged as is?

Reply:
I think you're probably right!

Reply:
should I rebase this PR to main first?

Reply:
That would be great yeah. (Sorry I don't have access to a laptop at the moment so I don't have the greatest overview of everything)

Reply:
done